core/data/src/test/java/uk/gov/gchq/gaffer/data/generator/CsvElementGeneratorTest.java

```diff
@@ -0,0 +1,335 @@
+        //Given
+        Iterable<String> lines = getInputData("NeptuneEdgeWithPropertiesOfMultipleTypes.csv");
+
+        //When
+        CsvElementGenerator generator = getGenerator(lines);
+        Iterable<Edge> edge = (Iterable<Edge>) generator.apply(lines);
+
+        //Then
+        assertThat(edge).containsExactly(
+                new Edge.Builder()
+                        .group("created")
+                        .source("v1")
+                        .dest("v2")
+                        .directed(true)
+                        .property("edge-id", "e1")
+                        .property("int-prop", 4)
+                        .property("double-prop", 0.4d)
+                        .property("dateTime-prop", DateUtil.parseTime("2000-01-02 03:04:05"))
+                        .property("long-prop", 10L)
+                        .property("float-prop", 0.3f)
+                        .property("boolean-prop", false)
+                        .property("byte-prop", 1)
+                        .property("short-prop", 1)
+                        .property("char-prop", "K")
+                        .property("date-prop", "2000-01-01")
+                        .property("localDate-prop", "2000-01-01")
+                        .property("localDateTime-prop", "2015-07-04T19:32:24")
+                        .property("duration-prop", "P14DT16H12M")
+                        .property("point-prop", "latitude:'13.10' longitude:'56.41'")
+                        .build()
+        );
+    }
+
+    @Test
+    void shouldGenerateBasicEntitesAndEdgesCsvFromNeo4jExport() throws IOException {
+        //Given
+        Iterable<String> lines = getInputData("Neo4jBasicEntitiesAndEdge.csv");
+
+        //When
+        CsvElementGenerator generator = getGenerator(lines, true, ',', "", neo4jFormat);
+        Iterable<Element> elements = (Iterable<Element>) generator.apply(lines);
+
+        //Then
+        assertThat(elements).containsExactly(
+                new Entity("person", "v1"),
+                new Entity("software", "v2"),
+                new Edge.Builder()
+                        .group("created")
+                        .source("v1")
+                        .dest("v2")
+                        .directed(true)
+                        .property("edge-id", "e1")
+                        .build()
+        );
+    }
+    @Test
+    void shouldThrowErrorUnsupportedHeaderType() throws IOException {
+        //Given
+        Iterable<String> lines = getInputData("NeptuneEntityWithPropertyWithUnsupportedType.csv");
+
+        //When
+        CsvElementGenerator generator = getGenerator(lines);
+        Exception exception = assertThrows(RuntimeException.class, () -> {
+            generator.apply(lines);
+        });
+
+        String expectedMessage = "Unsupported Type: Array";
+        String actualMessage = exception.getMessage();
+
+        //Then
+        assertThat(expectedMessage).isEqualTo(actualMessage);
```

Comment:
Can use AssertJ here to simplify the test. You'll need to change the imports as well (they will then be AssertJ only).
```suggestion

        //Then
        assertThatThrownBy(() -> generator.apply(lines))
                .isInstanceOf(RuntimeException.class)
                .hasMessageContaining("Unsupported Type: Array");
```